runtime/reconcilers/report.go

```diff
@@ -394,28 +397,77 @@ func (r *ReportReconciler) sendReport(ctx context.Context, self *runtimev1.Resou
 	}
 	defer release()
 
-	var ownerID string
+	var ownerID, explore, canvas, webOpenMode string
 	if id, ok := rep.Spec.Annotations["admin_owner_user_id"]; ok {
 		ownerID = id
 	}
+	if e, ok := rep.Spec.Annotations["explore"]; ok {
+		explore = e
+	}
+	if c, ok := rep.Spec.Annotations["canvas"]; ok {
+		canvas = c
+	}
+	if w, ok := rep.Spec.Annotations["web_open_mode"]; ok {
+		webOpenMode = w
+	} else {
+		webOpenMode = "none" // for older reports if web_open_path is not set
+		if path, ok := rep.Spec.Annotations["web_open_path"]; ok {
+			// parse path, extract explore name, it will be like /explore/{explore}
+			if explore == "" && strings.HasPrefix(path, "/explore/") {
+				explore = path[9:]
+				if explore[len(explore)-1] == '/' {
+					explore = explore[:len(explore)-1]
+				}
+			}
+		}
+		// still not found, try to extract mv from query args
+		if explore == "" && rep.Spec.QueryArgsJson != "" {
+			m := make(map[string]interface{})
+			err := json.Unmarshal([]byte(rep.Spec.QueryArgsJson), &m)
+			if err == nil {
+				if v, ok := m["metricsView"]; ok {
+					explore = v.(string)
+				} else if v, ok = m["metrics_view_name"]; ok {
+					explore = v.(string)
+				} else if v, ok = m["metrics_view"]; ok {
+					explore = v.(string)
+				}
+			}
+		}
+		if explore != "" {
+			webOpenMode = "recipient"
+		}
+	}
 
+	var reqWebOpenMode adminv1.ReportOptions_OpenMode
+	switch webOpenMode {
+	case "recipient":
```

Comment:
1. We really want to avoid a dependency on the `adminv1` package in the runtime if possible, except perhaps in the `runtime/drivers/admin` package. Trying to keep circular dependencies between the two services at a minimum. 
2. Can it just pass this through as an opaque `string` and let the admin service interpret it? That would also avoid repeating the string literals again in the cases here.